test/e2e/test/checks/monitoring.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package checks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	esClient "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// StackMonitoringChecks tests that the monitored resource pods have 3 containers ready and that there are documents indexed in the beat indexes
+// of the monitoring Elasticsearch clusters.
+type StackMonitoringChecks struct {
+	MonitoredNsn types.NamespacedName
+	Metrics      elasticsearch.Builder
+	Logs         elasticsearch.Builder
+	K            *test.K8sClient
+}
+
+func (c StackMonitoringChecks) Steps() test.StepList {
+	return test.StepList{
+		c.CheckBeatSidecars(),
+		c.CheckMetricbeatIndex(),
+		c.CheckFilebeatIndex(),
+	}
+}
+
+func (c StackMonitoringChecks) CheckBeatSidecars() test.Step {
+	return test.Step{
+		Name: "Check that beat sidecars are running",
+		Test: test.Eventually(func() error {
+			pods, err := c.K.GetPods(test.ESPodListOptions(c.MonitoredNsn.Namespace, c.MonitoredNsn.Name)...)
+			if err != nil {
+				return err
+			}
+			for _, pod := range pods {
+				if len(pod.Spec.Containers) != 3 {
+					return fmt.Errorf("expected %d containers, got %d", 3, len(pod.Spec.Containers))
+				}
+				if !k8s.IsPodReady(pod) {
+					return fmt.Errorf("pod %s not ready", pod.Name)
+				}
+			}
+			return nil
+		})}
+}
+
+func (c StackMonitoringChecks) CheckMetricbeatIndex() test.Step {
+	return test.Step{
+		Name: "Check that documents are indexed in one metricbeat-* index",
```

Comment:
I believe stack metrics are expected to be found in `.monitoring-kibana-` when `xpack.enabled` is set to `true`:

```
metricbeat-7.14.0-2021.07.12-000001 0 p STARTED  0   208b 10.55.208.22 test-kb-mon-metrics-lf6v-es-masterdata-1
metricbeat-7.14.0-2021.07.12-000001 0 r STARTED  0   208b 10.55.210.23 test-kb-mon-metrics-lf6v-es-masterdata-0
.monitoring-kibana-7-mb-2021.07.12  0 r STARTED  8 59.5kb 10.55.208.22 test-kb-mon-metrics-lf6v-es-masterdata-1
.monitoring-kibana-7-mb-2021.07.12  0 p STARTED  8 59.4kb 10.55.210.23 test-kb-mon-metrics-lf6v-es-masterdata-0
```